test/e2e/e2e_suite_test.go

```diff
@@ -24,10 +25,42 @@ import (
 func handleFlags() {
 	e2econfig.CopyFlags(e2econfig.Flags, flag.CommandLine)
 	framework.RegisterCommonFlags(flag.CommandLine)
-	framework.RegisterClusterFlags(flag.CommandLine)
+	/*
+		Using framework.RegisterClusterFlags(flag.CommandLine) results in a panic:
+		"flag redefined: kubeconfig".
+		This happens because controller-runtime registers the kubeconfig flag as well.
+		To solve this we set the framework's kubeconfig directly via the KUBECONFIG env var
+		instead of letting it call the flag. Since we also use the provider, num-nodes flags
+		which are handled manually.
+	*/
+	flag.StringVar(&framework.TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, skeleton (the fallback if not set), etc.)")
+	framework.TestContext.KubeConfig = os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+
+	flag.IntVar(&framework.TestContext.CloudConfig.NumNodes, "num-nodes", framework.DefaultNumNodes,
+		fmt.Sprintf("Number of nodes in the cluster. If the default value of '%q' is used the number of schedulable nodes is auto-detected.", framework.DefaultNumNodes))
+	flag.StringVar(&reportPath, "report-path", "/tmp/kind/logs", "the path to be used to dump test failure information")
 	flag.Parse()
 }
 
+var _ = ginkgo.BeforeSuite(func() {
+	// Make sure the framework's kubeconfig is set.
+	framework.ExpectNotEqual(framework.TestContext.KubeConfig, "", fmt.Sprintf("%s env var not set", clientcmd.RecommendedConfigPathEnvVar))
+
+	_, err := framework.LoadClientset()
+	framework.ExpectNoError(err)
+	_, err = framework.LoadConfig()
+	framework.ExpectNoError(err)
+
+	// TODO: Find a right way to fix nil issue for f.DumpAllNamespaceInfo
+	// when executing framework#dumpNamespaceInfo() upon test failure.
+	framework.TestContext.DumpLogsOnFailure = false
```

Comment:
So why do we need this then?

Reply:
if we don't do this, it would panic here https://github.com/kubernetes/kubernetes/blob/release-1.26/test/e2e/framework/framework.go#L274. This got fixed in later K8 versions (https://github.com/kubernetes/kubernetes/blob/release-1.27/test/e2e/framework/framework.go#L285-L287).

Reply:
I am guess I am confused here because `DumpLogsOnFailure` should already default to false and I don't see were that is being set to true elsewhere.

I see that `DumpAllNamespaceInfo` is being set here
https://github.com/kubernetes/kubernetes/blob/02b0b79eda5a3d41a07a6f2965ee97d492e8b6b2/test/e2e/framework/debug/init/init.go#L33-L35
but I am unsure if how could we use that directly.

Perhaps we can set it to to this ourselves
https://github.com/kubernetes/kubernetes/blob/02b0b79eda5a3d41a07a6f2965ee97d492e8b6b2/test/e2e/framework/debug/dump.go#L61

Reply:
It's set to true by default [here](https://github.com/kubernetes/kubernetes/blob/release-1.26/test/e2e/framework/test_context.go#L321), sure, now making `f.DumpAllNamespaceInfo `to use debug.`DumpAllNamespaceInfo` to get rid of this problem.